compiler/rustc_type_ir/src/sty.rs

```diff
@@ -231,6 +235,7 @@ const fn tykind_discriminant<I: Interner>(value: &TyKind<I>) -> usize {
         Placeholder(_) => 24,
         Infer(_) => 25,
         Error(_) => 26,
+        TyAlias(_, _) => 27,
```

Comment:
Hmm I was expecting this kind of thing would be where the legacy mangling causes different outputs, but I suppose that's not it, since existing variants don't get a different discriminant.